src/JustEat.StatsD/UdpTransport.cs

```diff
@@ -23,45 +19,31 @@ public UdpTransport(IPEndPointSource endPointSource)
 
         public void Send(string metric)
         {
-            Send(new[] {metric});
-        }
-
-        public void Send(IEnumerable<string> metrics)
-        {
-            var data = EventArgsPool.Pop();
-            //firehose alert! -- keep it moving!
-            if (data == null)
-            {
-                return;
-            }
-
-            data.RemoteEndPoint = _endpointSource.GetEndpoint();
-            data.SendPacketsElements = metrics.ToMaximumBytePackets()
-                .Select(bytes => new SendPacketsElement(bytes, 0, bytes.Length, true))
-                .ToArray();
+            var bytes = Encoding.UTF8.GetBytes(metric);
+            var endpoint = _endpointSource.GetEndpoint();
 
-            using (var udpClient = GetUdpClient())
+            using (var socket = GetUdpClient())
             {
-                udpClient.Client.Connect(data.RemoteEndPoint);
-                udpClient.Client.SendPacketsAsync(data);
+                socket.Connect(endpoint);
+                socket.Send(bytes);
             }
         }
 
-        public UdpClient GetUdpClient()
+        public Socket GetUdpClient()
```

Comment:
`CreateSocket()`?

Reply:
Might be worth cross-referencing this changes with the implementation I have internally that pools `UdpClient` (or is it `Socket`) that I created when profiling seemed to suggest that opening and closing lots of sockets created a fair bit of CPU overhead. I never got conclusive enough results in isolation to suggest it was a change worth making in general for the library.